src/ConnectWalletButton.tsx

```diff
@@ -29,27 +35,46 @@ const ConnectWalletButton = ({
     setIsDropdownOpen(false);
   };
 
-  const onConnectClick = async () => {
+  const onSwitchWalletClick = () => {
+    handleConnectClick();
+    setIsDropdownOpen(false);
+  };
+
+  const onConnectClick = () => {
     if (!walletAddress) {
-      await connectWallet();
+      handleConnectClick();
     }
   };
 
+  const showWalletIcon = icon && walletAddress;
+
   return (
     <div
       className={styles.walletButtonContainer}
       onMouseEnter={() => walletAddress && setIsDropdownOpen(true)}
       onMouseLeave={() => setIsDropdownOpen(false)}
     >
       <button onClick={onConnectClick} className={styles.walletButton}>
-        {displayedButtonText}
+        {showWalletIcon && (
+          <img
+            src={icon}
+            alt={`${walletName}-icon`}
+            className={styles.providerIcon}
+          />
+        )}
+        <span>{displayedButtonText}</span>
       </button>
 
       {isDropdownOpen && (
         <div className={styles.dropdown}>
           <div className={styles.dropdownItem} onClick={copyAddressToClipboard}>
             Copy Address
           </div>
+          {showSwitchWallet && (
+            <div className={styles.dropdownItem} onClick={onSwitchWalletClick}>
+              Switch Wallet
+            </div>
+          )}
```

Comment:
If a user only has one available wallet, don't show them the switch wallet option.